### ISSUE TITLE
A list of features to consider implementing

### ISSUE THREAD

**Issue by DineshSolanki** [DineshSolanki/FoliCon#9]
### This is a list of features that can be implemented.
 please comment on this issue for new feature requests. **Do not use it to tell about bugs or errors, for that please open a new issue.**
- [ ] Background watcher to automatically create poster Mode icons whenever a new media in the watched folder is added.

**Comment by ABOATDev** [comment]
Hello ! Nice soft ! 
- [ ] display the corresponding image, just before the text http://prntscr.com/t1l5rc

- [ ] works with video series

**Comment by DineshSolanki** [comment]
hey, thanks for suggesting, 
for the first feature, I initially worked on this, as I thought it would help in quick selection, but as the image will be downloaded for each result, it will be an unnecessary overhead, as an alternative, I can load images only of those which item is clicked but even then I'll have to send a request, so I don't find this feasible, correct me if there's something.

and for the second suggestion, I couldn't understand what you mean by video series, can you please explain more about it.

**Comment by DineshSolanki** [comment]
I tested the first feature today and I think it doesn't need to be an overhead as I only gave poster path to the selected item as a string, and image Control being smart enough automatically downloads it when needed so no need to request again

**Comment by DineshSolanki** [comment]
Poster preview in Poster mode result introduced in 0d84b5451d1fbd04765ed9d82363c7249283ebcb

**Comment by DineshSolanki** [comment]
I'm thinking of a new feature that will show files in the current searching folder, such as "Avengers endgame 480p 2019" beside the poster preview, it may help select correct title from result as you would see year from your media file. do you think I should implement it?

/polls Yes 'No'

**Comment by DineshSolanki** [comment]
About background watcher, I think it's not needed really, considering most of us don't want to let the app run in the background the whole time.

**Comment by pierx** [comment]
Regarding the new feature the custom icon setter, I think it would be nice if it target the exact single folder we would like to change and drag the exact icon directly into the box. 
Also it would be good if the desktop.ini IconResource= doesn't include the path but directly the icon name so the folder can be move freely, just like the normal icon mode.
Here's mockup for it
![folicon_request](https://user-images.githubusercontent.com/11854637/105280168-1cdf4f00-5be4-11eb-87cb-149cb2ec79c8.png)

**Comment by DineshSolanki** [comment]
@pierx Thank you for pointing that out, I have fixed that, also about the "one folder". you can delete other folders and icons keeping only the ones that you want, use the delete key. or other usual shortcuts such as "Shift + End" for selecting to end. 
![FoliCon single folder](https://user-images.githubusercontent.com/15937452/105525862-a3408180-5d07-11eb-8ea5-fb2b82a8b1ae.gif)

**Comment by DineshSolanki** [comment]
@pierx v3.2.1 available with "icon path fix"

**Comment by pierx** [comment]
> @pierx sorry for the late reply, I was a little occupied.
> Thank you for pointing that out, I have corrected that now,
> also about the "one folder". you can delete other folders and icons keeping only the ones that you want, use the delete key. or other usual shortcuts such as "Shift + End" for selecting to end.
> you can also rearrange the items.
> ![FoliCon single folder](https://user-images.githubusercontent.com/15937452/105525862-a3408180-5d07-11eb-8ea5-fb2b82a8b1ae.gif)

Thanks for the tips and the update!

**Comment by pierx** [comment]
Just a heads up, recently I'm facing issue with custom icon setter, where it doesn't response to selected or dragged folder. Apparently removing the Folicon folder in C:\Users\Username\AppData\Roaming fixed the issue.

**Comment by DineshSolanki** [comment]
feature -Ability to browse through and select a different poster in the poster mode.
added in version 3.3, looking forward for feedback.

**Comment by pierx** [comment]
Thanks for the update!
Suggestion, instead of clicking the poster for option, it might better to add button for that action.
![foli](https://user-images.githubusercontent.com/11854637/126479619-8ca30dc9-f3c9-4268-a051-aea0ffffd671.jpg)
By the way, this poster option doesn't seem to work on TV search mode.

**Comment by DineshSolanki** [comment]
> 
> 
> Thanks for the update!
> Suggestion, instead of clicking the poster for option, it might better to add button for that action.
> ![foli](https://user-images.githubusercontent.com/11854637/126479619-8ca30dc9-f3c9-4268-a051-aea0ffffd671.jpg)
> By the way, this poster option doesn't seem to work on TV search mode.

I Tried it on TV mode, It's working, but I'll take a look again at the possible causes, also this is yet not available for game mode.
about the button, Yes I should do, maybe next weekend,

**Comment by DineshSolanki** [comment]
@pierx TV mode issue is fixed, and show more button is also added. in the latest release, please download the same.
Also I want to take a moment to thank you for your valuable comments and suggestion, they are a big reason for FoliCon's evolution.

**Comment by pierx** [comment]
> @pierx TV mode issue is fixed, and show more button is also added in the latest release. please download the same.
> Also I want to take a moment to thank you for your valuable comments and suggestion, they are a big reason for FoliCon's evolution.

Thank you very much for the update. The show more button works nicely and the TV mode issue is fixed now.

**Comment by DineshSolanki** [comment]
Custom rating now also available, and more tool-tips.

**Comment by DineshSolanki** [comment]
4 new languages added, although in pre-release, needs more testing.

**Comment by DineshSolanki** [comment]
ID of picked titles now stored, for resolving titles automatically.

**Comment by riyaddl** [comment]
Hello, Great app. 

- [ ] the ability to add a custom overlay.
- [ ] the ability to change the rating background (with a star for example)

**Comment by DineshSolanki** [comment]
> Hello, Great app. 
> 
> - [ ] the ability to add a custom overlay.
> - [ ] the ability to change the rating badge (with a star for example)

Will look into them, in the meantime you might put here the custom overlays you fancy, so That I can aseess if they can be made into folicon

**Comment by tiwey15** [comment]
Love the app! I've been trying to find something like this for a while. Thank you for your hard work.

I do have two suggestions.

1) the DVD Icon is great, but itd be really cool if the icon could change based on the resolution of the file. I.e. if it's 4k replace "DVD Video" with "UHD" or 1080p with "HD" and everything else "SD".

2) Also, the name search funtion is really good, i wonder if it'd be too much trouble to have it rename the folder along with changing the icon.

Thanks so much. You are wonderful.

**Comment by DineshSolanki** [comment]
Hey, thanks,
About the 'DVD' on layout, that it not generated by 'FoliCon' it's an icon that I used from deviantArt, FoliCon would have trouble generating mockups like photoshop but I tried to get closer to that.
But we might be able to achieve if I manually modify icons with photoshop and try to identify quality from name of media file.

About the second one, renaming shouldn't be a problem, but there are tonnes of apps that do it better already, **I would recommend Filebot 4.7**

I'll add them as future enhancements -
1. Show media quality on icon
2. Feature to rename folders to correct media file.

**Comment by abewartech** [comment]
TinyMediaManager

**Comment by DineshSolanki** [comment]
> integrated with :
> 
>     * TinyMediaManager
> 
>     * Radarr & Sonarr

How should I go with that, AFAIK they don't have a plugin system, do you mean FoliCon should carry feature from them?

**Comment by TheFmC** [comment]
Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons

**Comment by DineshSolanki** [comment]
> Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons

Can you explain more, what was the mode you used and a screenshot showing the cover icon

**Comment by DineshSolanki** [comment]
> Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons

Hi @TheFmC Issue has been solved, thanks for reporting. 
Don't forget to STAR the repo✌️

**Comment by TheFmC** [comment]
> > Thank You For Your Work i really like the app .. but their is one problem iam facing .. low resolution for icons
> 
> Hi @TheFmC Issue has been solved, will be releasing soon, thanks for reporting.

Thank you for solving the problem in a short time
I can asure that it work now <3

**Comment by MasoudRahmani** [comment]
Anime? 🥹

**Comment by DineshSolanki** [comment]
> Anime? 🥹

Doesn't anime works with professional mode?

**Comment by MasoudRahmani** [comment]
> > Anime? 🥹
> 
> Doesn't anime works with professional mode?

Yes it does actually, also with data from IMDb, but i thought since it was not mentioned maybe there were more to it.
like an API [Anilist](https://docs.consumet.org/rest-api/Meta/anilist-anime/get-anime-info)

[Comic Books](https://metron.cloud/) ,

[Manga](https://docs.consumet.org/rest-api/Meta/anilist-manga/info)

**Comment by DineshSolanki** [comment]
> > > Anime? 🥹
> > 
> > Doesn't anime works with professional mode?
> 
> Yes it does actually, also with data from IMDb, but i thought since it was not mentioned maybe there were more to it.
> like an API [Anilist](https://docs.consumet.org/rest-api/Meta/anilist-anime/get-anime-info)
> 
> [Comic Books](https://metron.cloud/) ,
> 
> [Manga](https://docs.consumet.org/rest-api/Meta/anilist-manga/info) 

Will surely look into this information.

**Comment by DineshSolanki** [comment]
Pre-release available with dark mode, check it out, and help me get it release ready by your suggestions and feedback everyone https://github.com/DineshSolanki/FoliCon/releases/tag/v3.11.0-beta.1
